### ISSUE TITLE
Add validation checks for priors

### ISSUE THREAD

**Issue by danielinteractive** [Genentech/jmpost#72]
To do:

- [ ] Move assertions into validation checks for the priors
- [ ] both for the basic StanModule
- [ ] specific validation requirements on top of that for specific models
  - [ ] e.g. we need to have certain parameters as names in prior list for specific models

Dummy template example:

```r
.A <- setClass("A", slots = list(x = "numeric", y = "logical"))

a <- .A(x = c(1, 5), y = TRUE)

a

setValidity("A", function(object) {
  if (length(object@x) == 2) TRUE else "x has wrong length"
})

a2 <- .A(x = 3)
validObject(a)
a@x <- 3
validObject(a)

# getter method for x:
setGeneric(name = "x", def = function(object) standardGeneric("x"))

setMethod(
  f = "x", 
  signature = list(object = "A"),
  definition = function(object) {
    object@x
  }
)

x(a)

# setter method for x:
setGeneric(name = "x<-", def = function(object, value) standardGeneric("x<-"))

setReplaceMethod(
  f = "x",
  signature = "A",
  definition = function(object, value) {
    object@x <- value
    validObject(object)
    object
  }
)

x(a) <- c(daniel = 5, 6)
x(a)
a

x(a)["daniel"] <- 1
x(a)

x(a) <- 3

```

**Comment by shawnafang** [comment]
``` r
a<-b
```r

**Comment by danielinteractive** [comment]
@shawnafang can you execute the test line by line? which line does the error come from?

**Comment by shawnafang** [comment]
it comes with:
```r
priors(longmod)["eta_tilde_ks"] <- "normal(0,6)"
```
which report the error of validObject()

**Comment by danielinteractive** [comment]
hm I guess this could be a bug - is this on `main` branch? if yes please write a new issue for it

**Comment by danielinteractive** [comment]
@shawnafang cool I guess that is up to date (no new changes on main in the meantime) - then now next step would be to create an issue for this bug

**Comment by gowerc** [comment]
This is currently blocked by #90
